src/map.rs

```diff
@@ -97,602 +101,755 @@ pub enum MapEvent<T: Tile, C: Chunk<T>> {
+        self.chunk_dimensions
     }
 
-    fn insert_entity(&mut self, index: usize, entity: Entity) {
-        self.entities.insert(index, entity);
+    /// A Chunk's size in pixels.
+    pub fn chunk_size(&self) -> Vec2 {
+        Vec2::new(
+            self.tile_dimensions.width() * self.chunk_dimensions.width(),
+            self.tile_dimensions.height() * self.chunk_dimensions.height(),
+        )
     }
 
-    fn get_entity(&self, index: &usize) -> Option<&Entity> {
-        self.entities.get(index)
+    /// Returns the tile dimensions of the `TileMap`.
+    pub fn tile_dimensions(&self) -> Vec2 {
+        self.tile_dimensions
     }
 
-    fn events(&self) -> &Events<MapEvent<T, C>> {
-        &self.events
-    }
-
-    fn send_event(&mut self, event: MapEvent<T, C>) {
-        self.events.send(event);
-    }
-
-    fn events_update(&mut self) {
-        self.events.update()
-    }
-
-    fn events_reader(&mut self) -> EventReader<MapEvent<T, C>> {
-        self.events.get_reader()
-    }
-}
-
-impl<T: Tile, C: Chunk<T>> WorldMap<T, C> {
-    /// Returns a new WorldMap with the types `Tile` and `Chunk`.
-    pub fn new(dimensions: Vec2, texture_atlas: Handle<TextureAtlas>) -> WorldMap<T, C> {
-        let size = (dimensions.x() * dimensions.y()) as usize;
-        WorldMap {
-            dimensions,
-            handles: Vec::with_capacity(size),
-            entities: HashMap::default(),
-            events: Events::<MapEvent<T, C>>::default(),
-            texture_atlas,
-        }
+    /// Takes a `Tile` coordinate and returns its location in the `Map`.
+    pub fn chunk_to_world_coord(&self, coord: &Vec3, translation: Vec2) -> Option<Vec3> {
+        // takes in translation of tile
+        let chunk_x = (translation.x() / self.tile_dimensions.width() / self.dimensions.width())
+            + self.dimensions.max_x()
+            - 1.;
+        let chunk_y = 2.
+            - (translation.y() / self.tile_dimensions.height() / self.dimensions.height()
+                + self.dimensions.max_y()
+                - 1.);
+        let x = self.dimensions.width() * chunk_x + coord.x();
+        let y = (self.dimensions.height() * self.dimensions.max_y())
+            - (self.dimensions.height() * chunk_y)
+            + coord.y();
+        Some(Vec3::new(x, y, coord.z()))
     }
 }
 
-fn sprite_coord<T: Tile>(tile: &T, sprite_sheet_atlas: &TextureAtlas) -> Option<(usize, usize)> {
-    let sprite_idx = {
-        if let Some(handle) = tile.texture() {
-            sprite_sheet_atlas.get_texture_index(handle).unwrap()
-        } else {
-            return None;
-        }
-    };
-    let sprite_rect = sprite_sheet_atlas.textures[sprite_idx];
-    Some((sprite_rect.min.x() as usize, sprite_rect.min.y() as usize))
+/// A component bundle for `TileMap` entities.
+#[derive(Bundle, Debug)]
+pub struct TileMapComponents {
+    /// A `TileMap` which maintains chunks and its tiles.
+    pub tile_map: TileMap,
```

Comment:
I believe this will have to be `Handle<TileMap>` because, if I understand it correctly, the renderer will need to load the TileMap out of the assets store to use it. So you'll need to add the TileMap to an `Assets<TileMap` resource and put the handle that returns into the TileMapComponents. It's the same pattern that things like SpriteComponents use for materials and textures.